src/Configuration/Programmer/Programmer.cpp

```diff
@@ -22,17 +22,63 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "Programmer.h"
 
 #include <chrono>
+#include <iostream>
+#include <memory>
 #include <thread>
 
+#include "CFGCommon/CFGArg_auto.h"
 #include "CFGCommon/CFGCommon.h"
 
 void programmer_entry(const CFGCommon_ARG* cmdarg) {
-  CFG_POST_MSG("This is Programmer entry");
-  CFG_POST_MSG("  ProjectName : %s", cmdarg->projectName.c_str());
-  CFG_POST_MSG("  ProjectPath : %s", cmdarg->projectPath.c_str());
-  CFG_POST_MSG("  Device      : %s", cmdarg->device.c_str());
-  for (int i = 0; i < 20; i++) {
-    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
-    CFG_POST_MSG("Looping to test multithread - %d", i);
+  const std::string compilerName = cmdarg->compilerName;
+  if (compilerName == "N6FOEDAG8CompilerE") {
```

Comment:
this name is random and will be different for different compilers.

Reply:
Indeed, you are correct after I read further in the documentation. 
And now, I am thinking we should assign name to a compiler when it is instantiated.